### DISCUSSION TITLE
Delete/rename field values

### DISCUSSION THREAD

**Discussion by AStatkus** [msiemens/tinydb#390] [Q&A] [answered]
Hello,

I am new to tinydb, but is it possible to rename existing field with losing  all the data is assigned to it, and also is any easy way to delete a field with all its data like in mysql (ALTER TABLE table_name DROP COLUMN column_name;)

Thanks

**Comment by AStatkus** [comment]
Ok, how to delete i found :)

table.update(delete('name'))

But how to rename key i have no idea.

**Comment by msiemens** [comment, marked as answer]
I think you should be able to define a custom operation like this:

```python
def rename(from, to):
    def transform(doc):
        doc[to] = doc.pop(from, None)

    return transform
```

and use it just like `delete('name')`:

```python
table.update(rename('old_name', 'new_name'))
```

Does this work for you?

**Comment by AStatkus** [comment]
Thanks, it works, except i had to rename "from" to something, as it is not allowed as variable:)

But it works great.

**Reply by msiemens** [reply]
Oh, you're right! But I'm glad it works otherwise :)
